h2o-core/src/main/java/hex/CVModelBuilder.java

```diff
@@ -52,13 +52,19 @@ public void bulkBuildModels() {
             submodel_tasks[i] = H2O.submitTask(modelBuilders[i].trainModelImpl());
             if (++nRunning == parallelization) { //piece-wise advance in training the models
                 while (nRunning > 0) {
+                    final int waitForTaskIndex = i + 1 - nRunning;
                     try {
-                        int waitForTaskIndex = i + 1 - nRunning;
                         submodel_tasks[waitForTaskIndex].join();
                         finished(modelBuilders[waitForTaskIndex]);
                     } catch (RuntimeException t) {
-                        if (rt == null) rt = t;
+                        if (rt == null) {
+                            LOG.info("Exception from CV model #" + waitForTaskIndex + " will be reported as main exception.");
+                            rt = t;
+                        } else {
+                            LOG.warn("CV model #" + waitForTaskIndex + " failed, the exception will not be reported", t);
```

Comment:
out of scope question as it apparently used to work this way, but why do we wait for those loops to complete before throwing the exception.
Can't we also interrupt the loop (force nRunning = 0)
Same for the for-loop below.

Reply:
Afterthoughts.
We probably have to keep the loop going, but then it would be to try to abort the running tasks before waiting for their completion: can't we do that the same way we try to abort jobs?

Reply:
> can't we do that the same way we try to abort jobs?

We could absolutely do that. I am not sure what the motivation was here - maybe to give a cleaner clean-up workflow for the tasks that don't fail.

We cannot really say the caller can do anything with the succeded tasks anyway - there is no guarantee how many of them will be executed.

I think this could be revised. Also, the current way is not leveraging maximum parallelism right now: It starts 4 tasks and waits for them to finish, then it starts the last 5th task (which could really be running after one of the 4 finishes).

I think this is a candidate for a speedup. I also have an idea of running the main model alongside the CV tasks and let it advance when we know we can make additional iteration.